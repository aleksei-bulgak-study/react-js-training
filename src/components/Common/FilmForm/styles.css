.film-form .labeled-input,
.film-form .labeled-title,
.film-form .genres {
  display: block;
  text-transform: uppercase;
  color: var(--red);
  font-size: 1.2rem;
  margin-bottom: 1.5rem;
}

.film-form labeled-input:last-child {
  margin: 0;
}

.film-form labeled-input > * {
  display: block;
  width: 100%;
}

.film-form input,
.film-form select {
  line-height: 2rem;
  font-size: 1rem;
  background-color: #2e2e2e;
  color: var(--white);
  padding: 0.5rem 1rem;
  width: 100%;
  margin-top: 0.5rem;
}

.film-form__actions {
  margin-top: 2rem;
  display: flex;
  justify-content: flex-end;
  flex-wrap: wrap;
}

.film-form__actions__submit {
  margin-left: 1rem;
  margin-bottom: 1rem;
}

.film-form__actions__reset {
  margin-bottom: 1rem;
  padding: 0.5rem 3.5rem;
}

.film-form .labeled-input__error {
  font-size: 0.8rem;
  font-weight: 100;
  color: var(--red);
  text-transform: none;
}
