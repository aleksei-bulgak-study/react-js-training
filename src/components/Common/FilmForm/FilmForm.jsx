import React, { useCallback } from 'react';
import PropTypes from 'prop-types';
import { useFormik } from 'formik';
import * as Yup from 'yup';
import moment from 'moment';
import LabeledInput from '../LabeledInput';
import LabeledText from '../LabeledText';
import BaseModalWindow from '../../ModalWindows/BaseModalWindow';
import LabeledMultiSelect from '../LabeledMultiSelect';
import Button, { types } from '../Button';

import './styles.css';

const FilmForm = ({
  title,
  initialState = {},
  onSubmit,
  onClose,
  defaultGenres,
}) => {
  const onSave = useCallback((data) => onSubmit(data), [onSubmit]);

  const formik = useFormik({
    initialValues: initialState,
    validationSchema: Yup.object({
      title: Yup.string().required('Title is required field'),
      release_date: Yup.string()
        .transform((value) => {
          const dateValue = moment(value);

          return dateValue.isValid() ? dateValue.format('MM/dd/yyyy') : '';
        })
        .required('Release date is required field'),
      poster_path: Yup.string().url().required('Poster should be a valid url'),
      overview: Yup.string().required('Film description is required'),
      runtime: Yup.number()
        .min(0)
        .required('Runtime should be numeric value more then 0'),
      genres: Yup.array().min(1).required('At least 1 genre must be selected'),
    }),
    onSubmit: onSave,
  });

  const onSelectStateChange = useCallback(
    (values) => {
      const genres = values.map((item) => item.value);
      formik.setFieldValue('genres', genres, true);
    },
    [formik],
  );

  const getErrorMessageForField = useCallback(
    (fieldName) => {
      if (formik.touched[fieldName] && formik.errors[fieldName]) {
        return formik.errors[fieldName];
      }
      return '';
    },
    [formik],
  );

  return (
    <form className="film-form" onSubmit={formik.handleSubmit}>
      <BaseModalWindow title={title} onClose={onClose}>
        {formik.values.id !== undefined && (
          <LabeledText
            id="film-id"
            name="id"
            title="Movie id"
            value={formik.values.id}
          />
        )}
        <LabeledInput
          id="film-title"
          name="title"
          title="Title"
          value={formik.values.title}
          onChange={formik.handleChange}
          onBlur={formik.handleBlur}
          error={getErrorMessageForField('title')}
        />
        <LabeledInput
          id="film-release"
          name="release_date"
          title="Release date"
          type="date"
          value={formik.values.release_date}
          onChange={formik.handleChange}
          onBlur={formik.handleBlur}
          error={getErrorMessageForField('release_date')}
        />
        <LabeledInput
          id="film-url"
          name="poster_path"
          title="Movie url"
          type="url"
          value={formik.values.poster_path}
          onChange={formik.handleChange}
          onBlur={formik.handleBlur}
          error={getErrorMessageForField('poster_path')}
        />
        <LabeledMultiSelect
          title="genres"
          options={defaultGenres}
          onAction={onSelectStateChange}
          preselected={formik.values.genres}
          error={getErrorMessageForField('genres')}
        />
        <LabeledInput
          id="film-overview"
          name="overview"
          title="Overview"
          value={formik.values.overview}
          onChange={formik.handleChange}
          onBlur={formik.handleBlur}
          error={getErrorMessageForField('overview')}
        />
        <LabeledInput
          id="film-runtime"
          name="runtime"
          title="Runtime"
          type="number"
          value={formik.values.runtime}
          onChange={formik.handleChange}
          onBlur={formik.handleBlur}
          error={getErrorMessageForField('runtime')}
        />
        <div className="film-form__actions">
          <Button
            title="reset"
            onClick={formik.resetForm}
            type={types.SECONDARY}
            className="film-form__actions__reset"
          />
          <Button
            title="save"
            onClick={formik.handleSubmit}
            className="film-form__actions__submit"
          />
        </div>
      </BaseModalWindow>
    </form>
  );
};

FilmForm.propTypes = {
  initialState: PropTypes.shape({
    id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    title: PropTypes.string,
    release_date: PropTypes.string,
    poster_path: PropTypes.string,
    genres: PropTypes.arrayOf(PropTypes.string),
    overview: PropTypes.string,
    runtime: PropTypes.number,
  }),
  onClose: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
  title: PropTypes.string.isRequired,
  defaultGenres: PropTypes.arrayOf(
    PropTypes.shape({
      label: PropTypes.string.isRequired,
      value: PropTypes.string.isRequired,
    }),
  ).isRequired,
};

FilmForm.defaultProps = {
  initialState: {
    id: undefined,
    title: undefined,
    poster_path: undefined,
    release_date: undefined,
    genres: [],
    overview: undefined,
    runtime: undefined,
  },
};

export default FilmForm;
