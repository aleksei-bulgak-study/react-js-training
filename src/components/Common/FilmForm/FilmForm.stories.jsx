/* eslint-disable react/prop-types */
import React from 'react';
// eslint-disable-next-line import/no-extraneous-dependencies
import {
  withKnobs,
  text,
  number,
  object,
} from '@storybook/addon-knobs';
import FilmForm from './FilmForm';
import '../../../app.css';

export default {
  title: 'Form',
  decorators: [withKnobs],
};

const intialState = {
  id: number('Film Id', 1, { min: 1 }),
  title: `${text('Title', 'Test film title')}`,
  poster_path: `${text('Poster url', 'https://example.com')}`,
  release_date: `${text('Release date', '2020-11-15')}`,
  genres: ['Comedy'],
  overview: text('Description', 'some descritption'),
  runtime: number('Runtime', 12, { min: 1, max: 300 }),
};

const defaultGenres = object('Default genres', [
  { label: 'Comedy', value: 'Comedy' },
  { label: 'Horror', value: 'Horror' },
  { label: 'Drama', value: 'Drama' },
]);

export const Default = () => (
  <FilmForm
    title={text('Form Title', 'Add Film')}
    initialState={intialState}
    onSubmit={() => alert('Form was submitted')}
    onClose={() => alert('Form was closed')}
    defaultGenres={defaultGenres}
  />
);
