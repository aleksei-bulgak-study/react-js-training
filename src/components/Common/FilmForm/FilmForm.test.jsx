import React from 'react';
import { render, fireEvent, waitFor, cleanup } from '@testing-library/react';
import FilmForm from './FilmForm';

describe('FilmForm', () => {
  beforeEach(async () => {
    await cleanup();
    jest.resetAllMocks();
  });

  test('when all fields are empty then errors is displayed', async () => {
    const onSubmit = jest.fn();
    const onClose = jest.fn();
    const form = render(
      <FilmForm
        title="Add film"
        onSubmit={onSubmit}
        onClose={onClose}
        defaultGenres={[{ label: 'All', value: 'all' }]}
      />,
    );
    const submitButton = form.container.querySelector('.film-form');

    expect(
      form.container.querySelectorAll('p.labeled-input__error').length,
    ).toEqual(0);

    await waitFor(() => {
      fireEvent.submit(submitButton);
    });

    expect(
      form.container.querySelectorAll('p.labeled-input__error').length,
    ).toEqual(5);

    expect(onSubmit).not.toHaveBeenCalled();
    expect(form).toMatchSnapshot();
  });

  test('when all fields are populated then onSubmit is called', async () => {
    const onSubmit = jest.fn();
    const onClose = jest.fn();
    const form = render(
      <FilmForm
        title="Add film"
        initialState={{ genres: ['all'] }}
        onSubmit={onSubmit}
        onClose={onClose}
        defaultGenres={[{ label: 'All', value: 'all' }]}
      />,
    );
    const submitButton = form.container.querySelector('.film-form__actions__submit');

    await waitFor(() => {
      fireEvent.change(form.container.querySelector('#film-title'), {
        target: { value: 'test title' },
      });
    });
    await waitFor(() => {
      fireEvent.change(form.container.querySelector('#film-release'), {
        target: { value: '11/10/2020' },
      });
    });
    await waitFor(() => {
      fireEvent.change(form.container.querySelector('#film-url'), {
        target: { value: 'https://example.com' },
      });
    });
    await waitFor(() => {
      fireEvent.change(form.container.querySelector('#film-overview'), {
        target: { value: 'test description' },
      });
    });
    await waitFor(() => {
      fireEvent.change(form.container.querySelector('#film-runtime'), {
        target: { value: 123 },
      });
    });

    await waitFor(() => {
      fireEvent.submit(submitButton);
    });

    expect(
      form.container.querySelectorAll('p.labeled-input__error').length,
    ).toEqual(0);
    expect(form.container.querySelectorAll('.genres__error').length).toEqual(0);

    expect(onSubmit).toHaveBeenCalled();
    expect(form).toMatchSnapshot();
  });

  test('when all fields are populated then onReset clicked then values resetted', async () => {
    const onSubmit = jest.fn();
    const onClose = jest.fn();
    const form = render(
      <FilmForm
        title="Add film"
        initialState={{ genres: ['all'] }}
        onSubmit={onSubmit}
        onClose={onClose}
        defaultGenres={[{ label: 'All', value: 'all' }]}
      />,
    );
    const resetButton = form.container.querySelector(
      '.film-form__actions__reset',
    );

    await waitFor(() => {
      fireEvent.change(form.container.querySelector('#film-title'), {
        target: { value: 'test title' },
      });
    });
    await waitFor(() => {
      fireEvent.change(form.container.querySelector('#film-release'), {
        target: { value: '11/10/2020' },
      });
    });
    await waitFor(() => {
      fireEvent.change(form.container.querySelector('#film-url'), {
        target: { value: 'https://example.com' },
      });
    });
    await waitFor(() => {
      fireEvent.change(form.container.querySelector('#film-overview'), {
        target: { value: 'test description' },
      });
    });
    await waitFor(() => {
      fireEvent.change(form.container.querySelector('#film-runtime'), {
        target: { value: 123 },
      });
    });

    expect(
      form.container.querySelectorAll('p.labeled-input__error').length,
    ).toEqual(0);
    expect(form.container.querySelectorAll('.genres__error').length).toEqual(0);

    await waitFor(() => {
      fireEvent.click(resetButton);
    });

    expect(form.container.querySelector('#film-title').value).toEqual('');
    expect(form.container.querySelector('#film-release').value).toEqual('');
    expect(form.container.querySelector('#film-url').value).toEqual('');
    expect(form.container.querySelector('#film-overview').value).toEqual('');
    expect(form.container.querySelector('#film-runtime').value).toEqual('');

    expect(onSubmit).not.toHaveBeenCalled();
    expect(form).toMatchSnapshot();
  });
});
