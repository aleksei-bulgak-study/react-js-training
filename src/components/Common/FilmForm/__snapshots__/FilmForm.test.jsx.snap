// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmForm when all fields are empty then errors is displayed 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        class="film-form"
      >
        <div
          class="modal-window__overlay"
        />
        <div
          class="modal-window"
        >
          <div
            class="modal-window__wrapper "
          >
            <h1
              class="modal-window__title"
            >
              Add film
            </h1>
            <button
              class="close-button modal-window__close large"
              type="button"
            >
              Close
            </button>
            <div
              class="labeled-input"
            >
              <label
                class="labeled-input__input"
                for="film-title"
              >
                Title
                <input
                  id="film-title"
                  name="title"
                  type="text"
                  value=""
                />
              </label>
              <p
                class="labeled-input__error"
              >
                Title is required field
              </p>
            </div>
            <div
              class="labeled-input"
            >
              <label
                class="labeled-input__input"
                for="film-release"
              >
                Release date
                <input
                  id="film-release"
                  name="release_date"
                  type="date"
                  value=""
                />
              </label>
              <p
                class="labeled-input__error"
              >
                Release date is required field
              </p>
            </div>
            <div
              class="labeled-input"
            >
              <label
                class="labeled-input__input"
                for="film-url"
              >
                Movie url
                <input
                  id="film-url"
                  name="poster_path"
                  type="url"
                  value=""
                />
              </label>
              <p
                class="labeled-input__error"
              >
                Poster should be a valid url
              </p>
            </div>
            <label
              class="genres"
              for="genres"
            >
              genres
              <div
                class="go2139093995 multi-select"
              >
                <div
                  aria-expanded="false"
                  aria-readonly="true"
                  class="go2646822163 dropdown-container"
                  tabindex="0"
                >
                  <div
                    class="go2642161244 dropdown-heading"
                  >
                    <div
                      class="dropdown-heading-value"
                    >
                      <span
                        class="gray"
                      >
                        Select...
                      </span>
                    </div>
                    <svg
                      class="dropdown-heading-dropdown-arrow gray"
                      fill="none"
                      height="24"
                      stroke="currentColor"
                      stroke-width="2"
                      width="24"
                    >
                      <path
                        d="M6 9L12 15 18 9"
                      />
                    </svg>
                  </div>
                </div>
              </div>
              <p
                class="genres__error"
              >
                genres field must have at least 1 items
              </p>
            </label>
            <div
              class="labeled-input"
            >
              <label
                class="labeled-input__input"
                for="film-overview"
              >
                Overview
                <input
                  id="film-overview"
                  name="overview"
                  type="text"
                  value=""
                />
              </label>
              <p
                class="labeled-input__error"
              >
                Film description is required
              </p>
            </div>
            <div
              class="labeled-input"
            >
              <label
                class="labeled-input__input"
                for="film-runtime"
              >
                Runtime
                <input
                  id="film-runtime"
                  name="runtime"
                  type="number"
                  value=""
                />
              </label>
              <p
                class="labeled-input__error"
              >
                Runtime should be numeric value more then 0
              </p>
            </div>
            <div
              class="film-form__actions"
            >
              <button
                class="button secondary film-form__actions__reset"
                type="button"
              >
                reset
              </button>
              <button
                class="button film-form__actions__submit"
                type="button"
              >
                save
              </button>
            </div>
          </div>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      class="film-form"
    >
      <div
        class="modal-window__overlay"
      />
      <div
        class="modal-window"
      >
        <div
          class="modal-window__wrapper "
        >
          <h1
            class="modal-window__title"
          >
            Add film
          </h1>
          <button
            class="close-button modal-window__close large"
            type="button"
          >
            Close
          </button>
          <div
            class="labeled-input"
          >
            <label
              class="labeled-input__input"
              for="film-title"
            >
              Title
              <input
                id="film-title"
                name="title"
                type="text"
                value=""
              />
            </label>
            <p
              class="labeled-input__error"
            >
              Title is required field
            </p>
          </div>
          <div
            class="labeled-input"
          >
            <label
              class="labeled-input__input"
              for="film-release"
            >
              Release date
              <input
                id="film-release"
                name="release_date"
                type="date"
                value=""
              />
            </label>
            <p
              class="labeled-input__error"
            >
              Release date is required field
            </p>
          </div>
          <div
            class="labeled-input"
          >
            <label
              class="labeled-input__input"
              for="film-url"
            >
              Movie url
              <input
                id="film-url"
                name="poster_path"
                type="url"
                value=""
              />
            </label>
            <p
              class="labeled-input__error"
            >
              Poster should be a valid url
            </p>
          </div>
          <label
            class="genres"
            for="genres"
          >
            genres
            <div
              class="go2139093995 multi-select"
            >
              <div
                aria-expanded="false"
                aria-readonly="true"
                class="go2646822163 dropdown-container"
                tabindex="0"
              >
                <div
                  class="go2642161244 dropdown-heading"
                >
                  <div
                    class="dropdown-heading-value"
                  >
                    <span
                      class="gray"
                    >
                      Select...
                    </span>
                  </div>
                  <svg
                    class="dropdown-heading-dropdown-arrow gray"
                    fill="none"
                    height="24"
                    stroke="currentColor"
                    stroke-width="2"
                    width="24"
                  >
                    <path
                      d="M6 9L12 15 18 9"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <p
              class="genres__error"
            >
              genres field must have at least 1 items
            </p>
          </label>
          <div
            class="labeled-input"
          >
            <label
              class="labeled-input__input"
              for="film-overview"
            >
              Overview
              <input
                id="film-overview"
                name="overview"
                type="text"
                value=""
              />
            </label>
            <p
              class="labeled-input__error"
            >
              Film description is required
            </p>
          </div>
          <div
            class="labeled-input"
          >
            <label
              class="labeled-input__input"
              for="film-runtime"
            >
              Runtime
              <input
                id="film-runtime"
                name="runtime"
                type="number"
                value=""
              />
            </label>
            <p
              class="labeled-input__error"
            >
              Runtime should be numeric value more then 0
            </p>
          </div>
          <div
            class="film-form__actions"
          >
            <button
              class="button secondary film-form__actions__reset"
              type="button"
            >
              reset
            </button>
            <button
              class="button film-form__actions__submit"
              type="button"
            >
              save
            </button>
          </div>
        </div>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`FilmForm when all fields are populated then onReset clicked then values resetted 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        class="film-form"
      >
        <div
          class="modal-window__overlay"
        />
        <div
          class="modal-window"
        >
          <div
            class="modal-window__wrapper "
          >
            <h1
              class="modal-window__title"
            >
              Add film
            </h1>
            <button
              class="close-button modal-window__close large"
              type="button"
            >
              Close
            </button>
            <div
              class="labeled-input"
            >
              <label
                class="labeled-input__input"
                for="film-title"
              >
                Title
                <input
                  id="film-title"
                  name="title"
                  type="text"
                  value=""
                />
              </label>
            </div>
            <div
              class="labeled-input"
            >
              <label
                class="labeled-input__input"
                for="film-release"
              >
                Release date
                <input
                  id="film-release"
                  name="release_date"
                  type="date"
                  value=""
                />
              </label>
            </div>
            <div
              class="labeled-input"
            >
              <label
                class="labeled-input__input"
                for="film-url"
              >
                Movie url
                <input
                  id="film-url"
                  name="poster_path"
                  type="url"
                  value=""
                />
              </label>
            </div>
            <label
              class="genres"
              for="genres"
            >
              genres
              <div
                class="go2139093995 multi-select"
              >
                <div
                  aria-expanded="false"
                  aria-readonly="true"
                  class="go2646822163 dropdown-container"
                  tabindex="0"
                >
                  <div
                    class="go2642161244 dropdown-heading"
                  >
                    <div
                      class="dropdown-heading-value"
                    >
                      <span>
                        All items are selected.
                      </span>
                    </div>
                    <svg
                      class="dropdown-heading-dropdown-arrow gray"
                      fill="none"
                      height="24"
                      stroke="currentColor"
                      stroke-width="2"
                      width="24"
                    >
                      <path
                        d="M6 9L12 15 18 9"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </label>
            <div
              class="labeled-input"
            >
              <label
                class="labeled-input__input"
                for="film-overview"
              >
                Overview
                <input
                  id="film-overview"
                  name="overview"
                  type="text"
                  value=""
                />
              </label>
            </div>
            <div
              class="labeled-input"
            >
              <label
                class="labeled-input__input"
                for="film-runtime"
              >
                Runtime
                <input
                  id="film-runtime"
                  name="runtime"
                  type="number"
                  value=""
                />
              </label>
            </div>
            <div
              class="film-form__actions"
            >
              <button
                class="button secondary film-form__actions__reset"
                type="button"
              >
                reset
              </button>
              <button
                class="button film-form__actions__submit"
                type="button"
              >
                save
              </button>
            </div>
          </div>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      class="film-form"
    >
      <div
        class="modal-window__overlay"
      />
      <div
        class="modal-window"
      >
        <div
          class="modal-window__wrapper "
        >
          <h1
            class="modal-window__title"
          >
            Add film
          </h1>
          <button
            class="close-button modal-window__close large"
            type="button"
          >
            Close
          </button>
          <div
            class="labeled-input"
          >
            <label
              class="labeled-input__input"
              for="film-title"
            >
              Title
              <input
                id="film-title"
                name="title"
                type="text"
                value=""
              />
            </label>
          </div>
          <div
            class="labeled-input"
          >
            <label
              class="labeled-input__input"
              for="film-release"
            >
              Release date
              <input
                id="film-release"
                name="release_date"
                type="date"
                value=""
              />
            </label>
          </div>
          <div
            class="labeled-input"
          >
            <label
              class="labeled-input__input"
              for="film-url"
            >
              Movie url
              <input
                id="film-url"
                name="poster_path"
                type="url"
                value=""
              />
            </label>
          </div>
          <label
            class="genres"
            for="genres"
          >
            genres
            <div
              class="go2139093995 multi-select"
            >
              <div
                aria-expanded="false"
                aria-readonly="true"
                class="go2646822163 dropdown-container"
                tabindex="0"
              >
                <div
                  class="go2642161244 dropdown-heading"
                >
                  <div
                    class="dropdown-heading-value"
                  >
                    <span>
                      All items are selected.
                    </span>
                  </div>
                  <svg
                    class="dropdown-heading-dropdown-arrow gray"
                    fill="none"
                    height="24"
                    stroke="currentColor"
                    stroke-width="2"
                    width="24"
                  >
                    <path
                      d="M6 9L12 15 18 9"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </label>
          <div
            class="labeled-input"
          >
            <label
              class="labeled-input__input"
              for="film-overview"
            >
              Overview
              <input
                id="film-overview"
                name="overview"
                type="text"
                value=""
              />
            </label>
          </div>
          <div
            class="labeled-input"
          >
            <label
              class="labeled-input__input"
              for="film-runtime"
            >
              Runtime
              <input
                id="film-runtime"
                name="runtime"
                type="number"
                value=""
              />
            </label>
          </div>
          <div
            class="film-form__actions"
          >
            <button
              class="button secondary film-form__actions__reset"
              type="button"
            >
              reset
            </button>
            <button
              class="button film-form__actions__submit"
              type="button"
            >
              save
            </button>
          </div>
        </div>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`FilmForm when all fields are populated then onSubmit is called 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        class="film-form"
      >
        <div
          class="modal-window__overlay"
        />
        <div
          class="modal-window"
        >
          <div
            class="modal-window__wrapper "
          >
            <h1
              class="modal-window__title"
            >
              Add film
            </h1>
            <button
              class="close-button modal-window__close large"
              type="button"
            >
              Close
            </button>
            <div
              class="labeled-input"
            >
              <label
                class="labeled-input__input"
                for="film-title"
              >
                Title
                <input
                  id="film-title"
                  name="title"
                  type="text"
                  value="test title"
                />
              </label>
            </div>
            <div
              class="labeled-input"
            >
              <label
                class="labeled-input__input"
                for="film-release"
              >
                Release date
                <input
                  id="film-release"
                  name="release_date"
                  type="date"
                  value="2020-10-30"
                />
              </label>
            </div>
            <div
              class="labeled-input"
            >
              <label
                class="labeled-input__input"
                for="film-url"
              >
                Movie url
                <input
                  id="film-url"
                  name="poster_path"
                  type="url"
                  value="https://example.com"
                />
              </label>
            </div>
            <label
              class="genres"
              for="genres"
            >
              genres
              <div
                class="go2139093995 multi-select"
              >
                <div
                  aria-expanded="false"
                  aria-readonly="true"
                  class="go2646822163 dropdown-container"
                  tabindex="0"
                >
                  <div
                    class="go2642161244 dropdown-heading"
                  >
                    <div
                      class="dropdown-heading-value"
                    >
                      <span>
                        All items are selected.
                      </span>
                    </div>
                    <svg
                      class="dropdown-heading-dropdown-arrow gray"
                      fill="none"
                      height="24"
                      stroke="currentColor"
                      stroke-width="2"
                      width="24"
                    >
                      <path
                        d="M6 9L12 15 18 9"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </label>
            <div
              class="labeled-input"
            >
              <label
                class="labeled-input__input"
                for="film-overview"
              >
                Overview
                <input
                  id="film-overview"
                  name="overview"
                  type="text"
                  value="test description"
                />
              </label>
            </div>
            <div
              class="labeled-input"
            >
              <label
                class="labeled-input__input"
                for="film-runtime"
              >
                Runtime
                <input
                  id="film-runtime"
                  name="runtime"
                  type="number"
                  value="123"
                />
              </label>
            </div>
            <div
              class="film-form__actions"
            >
              <button
                class="button secondary film-form__actions__reset"
                type="button"
              >
                reset
              </button>
              <button
                class="button film-form__actions__submit"
                type="button"
              >
                save
              </button>
            </div>
          </div>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      class="film-form"
    >
      <div
        class="modal-window__overlay"
      />
      <div
        class="modal-window"
      >
        <div
          class="modal-window__wrapper "
        >
          <h1
            class="modal-window__title"
          >
            Add film
          </h1>
          <button
            class="close-button modal-window__close large"
            type="button"
          >
            Close
          </button>
          <div
            class="labeled-input"
          >
            <label
              class="labeled-input__input"
              for="film-title"
            >
              Title
              <input
                id="film-title"
                name="title"
                type="text"
                value="test title"
              />
            </label>
          </div>
          <div
            class="labeled-input"
          >
            <label
              class="labeled-input__input"
              for="film-release"
            >
              Release date
              <input
                id="film-release"
                name="release_date"
                type="date"
                value="2020-10-30"
              />
            </label>
          </div>
          <div
            class="labeled-input"
          >
            <label
              class="labeled-input__input"
              for="film-url"
            >
              Movie url
              <input
                id="film-url"
                name="poster_path"
                type="url"
                value="https://example.com"
              />
            </label>
          </div>
          <label
            class="genres"
            for="genres"
          >
            genres
            <div
              class="go2139093995 multi-select"
            >
              <div
                aria-expanded="false"
                aria-readonly="true"
                class="go2646822163 dropdown-container"
                tabindex="0"
              >
                <div
                  class="go2642161244 dropdown-heading"
                >
                  <div
                    class="dropdown-heading-value"
                  >
                    <span>
                      All items are selected.
                    </span>
                  </div>
                  <svg
                    class="dropdown-heading-dropdown-arrow gray"
                    fill="none"
                    height="24"
                    stroke="currentColor"
                    stroke-width="2"
                    width="24"
                  >
                    <path
                      d="M6 9L12 15 18 9"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </label>
          <div
            class="labeled-input"
          >
            <label
              class="labeled-input__input"
              for="film-overview"
            >
              Overview
              <input
                id="film-overview"
                name="overview"
                type="text"
                value="test description"
              />
            </label>
          </div>
          <div
            class="labeled-input"
          >
            <label
              class="labeled-input__input"
              for="film-runtime"
            >
              Runtime
              <input
                id="film-runtime"
                name="runtime"
                type="number"
                value="123"
              />
            </label>
          </div>
          <div
            class="film-form__actions"
          >
            <button
              class="button secondary film-form__actions__reset"
              type="button"
            >
              reset
            </button>
            <button
              class="button film-form__actions__submit"
              type="button"
            >
              save
            </button>
          </div>
        </div>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
